File: src/components/JSONHandler/src/DeleteSubMenu.cpp
#include "JSONHandler/DeleteSubMenu.h"
#include "JSONHandler/RPC2Marshaller.h"

using namespace RPC2Communication;

DeleteSubMenu::DeleteSubMenu()
:RPC2Request( RPC2Marshaller :: METHOD_DELETESUBMENU_REQUEST )
{}
 
DeleteSubMenu::~DeleteSubMenu()
{}

void DeleteSubMenu::setMenuId(unsigned int menuId)
{
    this->menuId = menuId;
}
        
unsigned int DeleteSubMenu::getMenuId() const
{
    return menuId;
}

File: mobile/android/SyncProxyAndroid/src/com/ford/syncV4/proxy/rpc/DeleteSubMenu.java
package com.ford.syncV4.proxy.rpc;

import java.util.Hashtable;

import com.ford.syncV4.proxy.RPCRequest;
import com.ford.syncV4.proxy.constants.Names;

public class DeleteSubMenu extends RPCRequest {

    public DeleteSubMenu() {
        super("DeleteSubMenu");
    }
    public DeleteSubMenu(Hashtable hash) {
        super(hash);
    }
    public Integer getMenuID() {
        return (Integer) parameters.get( Names.menuID );
    }
    public void setMenuID( Integer menuID ) {
        if (menuID != null) {
            parameters.put(Names.menuID, menuID );
        }
    }
}